DataConnectors/ESET Enterprise Inspector/datacollector/__init__.py

```diff
@@ -0,0 +1,48 @@
+# Taken from https://docs.microsoft.com/azure/azure-monitor/platform/data-collector-api
+
+import json
+import requests
+import datetime
+import hashlib
+import hmac
+import base64
+
+from enterpriseinspector.eifunctions import exit_error
+
+#####################
+######Functions######  
+#####################
+
+# Build the API signature
+def build_signature(customer_id, shared_key, date, content_length, method, content_type, resource):
+    x_headers = 'x-ms-date:' + date
+    string_to_hash = method + "\n" + str(content_length) + "\n" + content_type + "\n" + x_headers + "\n" + resource
+    bytes_to_hash = bytes(string_to_hash, encoding="utf-8")  
+    decoded_key = base64.b64decode(shared_key)
+    encoded_hash = base64.b64encode(hmac.new(decoded_key, bytes_to_hash, digestmod=hashlib.sha256).digest()).decode()
+    authorization = "SharedKey {}:{}".format(customer_id,encoded_hash)
+    return authorization
+
+# Build and send a request to the POST API
+def post_data(customer_id, shared_key, body, log_type):
+    method = 'POST'
+    content_type = 'application/json'
+    resource = '/api/logs'
+    rfc1123date = datetime.datetime.utcnow().strftime('%a, %d %b %Y %H:%M:%S GMT')
+    content_length = len(body)
+    signature = build_signature(customer_id, shared_key, rfc1123date, content_length, method, content_type, resource)
+    uri = 'https://' + customer_id + '.ods.opinsights.azure.com' + resource + '?api-version=2016-04-01'
```

Comment:
Is it an issue to hard-code the version number?

Reply:
I'm totally open to changing this. However, the documentation also hard-codes it: https://docs.microsoft.com/en-us/azure/azure-monitor/platform/data-collector-api.

But please let me know if you have any other suggestions.

Reply:
@dmaasland , since it is in the template, we will take as it is. Thanks!